### ISSUE TITLE
Use smarter defaults in type specification

### ISSUE THREAD

**Issue by andylowry** [RepreZen/KaiZen-OpenApi-Parser#124]
It's pretty annoying to constantly have to write stuff like this in the types specification:

```
    description:
       name: Description
       type: String
```

The property key above is the field's `id` value, and is the default for it path within its containing object. The `name` should also have an intelligent default based on the `id`, and settling on `String` as the default would make things additionally concise.

For name default, the following rules will apply:

* Only the final component of the `id` value will be considered (where components are separated by slash characters). E.g. with `components/schemas` only `schemas` will be considered when determining the default `name`.

* If the field is a scalar, or if the `id` value does not end in `s`, then the last component of the `id`, with its first character upper-cased, is the default `name`.
* Otherwise (collection or map field, and `id` ends in `s`), the last component of the `id`, with the first character upper-cased and the final `s` removed, is the default name.

**Comment by andylowry** [comment]
Implemented in #125

**Comment by andylowry** [comment]
As described in #125, defaulting a missing type to "String" was altered to retain existing default behavior (which I had forgotten about). Rules for a missing field type are now:

* If the field name is the name of a type defined in the type config, that type is used for the field
* Otherwise, `String` is used.
